library/alloc/src/vec.rs

```diff
@@ -2928,6 +2934,35 @@ impl<T> IntoIter<T> {
         self.ptr = self.buf.as_ptr();
         self.end = self.buf.as_ptr();
     }
+
+    /// Shifts the remaining elements to `offset` and then converts the whole allocation into a Vec
+    /// with `vec.len() == offset + self.len()`
+    ///
+    /// # Safety
+    ///
+    /// When a non-zero offset is passed the resulting Vec will have an uninitialized prefix
+    /// that needs to be filled before the Vec is valid again.
+    ///
+    /// * `offset + self.len()` must not exceed `self.cap`
+    /// * `offset == 0` is always valid
+    /// * `offset` must be positive
+    unsafe fn into_vec(self, offset: isize) -> Vec<T> {
```

Comment:
I think this shouldn't be called `into_vec`, slice have `into_vec` but does not take in parameter, this would make people confuse since the other `into_vec` does not take in an offset.

Reply:
It's a private method, only available in the vec module, so I think it should be fine.

Reply:
I know it is a private method but it will confuse those people first time seeing this when they know about the public method `into_vec`, I believe at least we should make it easier to new contributors with least confusion.